xds/src/main/java/io/grpc/xds/sds/TlsContextManager.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2019 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.xds.sds;
+
+import io.envoyproxy.envoy.api.v2.auth.DownstreamTlsContext;
+import io.envoyproxy.envoy.api.v2.auth.UpstreamTlsContext;
+import io.grpc.Internal;
+import io.netty.handler.ssl.SslContext;
+
+/**
+ * Class to manage secrets used to create SSL contexts - this effectively manages SSL contexts
+ * (aka TlsContexts) based on inputs we get from xDS. This is used by gRPC-xds to access the
+ * SSL contexts/secrets and is not public API.
+ * TODO(sanjaypujare): flesh out the implementation and remove the old {@link SecretManager}
+ * once done.
+ */
+@Internal
+public class TlsContextManager {
+
+  public SecretProvider<SslContext> findOrCreateServerSslContextProvider(
+      DownstreamTlsContext downstreamTlsContext) {
+    // TODO(sanjaypujare): implement required SecrerProvider
```

Comment:
SecretProvider? also L42

Reply:
yes I meant the required implementation of SecretProvider similar to the current `TlsCertificateSecretVolumeSecretProvider` that implements `SecretProvider<TlsCertificateStore>`. Will modify the comment to be clearer

Reply:
oops should be more specific, i meant typo.

Reply:
fixed typo as well